src/box/lua/experimental/connpool.lua

```diff
@@ -54,6 +54,45 @@ local function connect_using_config(instance_name, opts)
     return conn
 end
 
+local function is_labels_match(labels, instance_labels)
+    if labels == nil then
+        return true
+    end
+    if instance_labels == nil then
+        return false
+    end
+    for _, label in pairs(labels) do
+        if instance_labels[label] == nil then
+            return false
+        end
+    end
```

Comment:
As I see, we check a presence of the label, but we don't verify that it has a value given by a caller.

It is easy to imagine that one have the `zone` label and wants to filter instances from a specific zone.

I propose to implement the `labels` filter with the following semantic.

The instance `I` is accepted by the `opts.labels` filter if `I.labels` is *covered* by `opts.labels`. It means that each key-value pair from `opts.labels` has the same key-value pair in `I.labels`.

If we will need a filter that verifies a presence of a label without taking into account its value, I would add it later with a name like `has_labels`.

```lua
pool.get_candidates({
    has_labels = {'mylabel1', 'mylabel2'},
})
```